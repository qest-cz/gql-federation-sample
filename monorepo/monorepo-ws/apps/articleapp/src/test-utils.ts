import { testServer} from '@monorepo-ws/server'; 
import { modules } from './schema/modules';
import { gql } from 'apollo-server';
import { GqlUser } from './schema/interfaces';

const testModules = [
    {
        typeDefs: gql`
            extend type Query{
            getUser: User!
           } 
        `,
        resolvers: {
            Query: {
                getUser: (): Pick<GqlUser, 'id'> => {
                    return {id: 1}
                }
            }
        }
    }
]

export const server = testServer([...modules, ...testModules])