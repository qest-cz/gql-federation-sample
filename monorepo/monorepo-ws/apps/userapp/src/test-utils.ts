import { gql } from 'apollo-server';
import { testServer} from '@monorepo-ws/server'; 
import { modules } from './schema/modules';

export const server = testServer(modules)
