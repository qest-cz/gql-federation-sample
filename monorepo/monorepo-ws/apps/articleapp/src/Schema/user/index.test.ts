import { server } from "@qest/express-utils"
import { gql } from "apollo-server"
import { server as testServer } from "../../test-utils"

const query = gql`
    query {
        getUser {
            id,
            articles {
                id,
                title,
                authorId
            }
        }
    }
`


describe('user resolver', () => {
    it('get article of user', async () => {
        const result = await testServer.executeOperation({
            query: query
        })
        console.log(JSON.stringify(result.data))
        expect(result.data).toMatchSnapshot({getUser: userArticleMatcher})
    })
})

const userArticleMatcher = {
    id: expect.any(String),
    articles: [
        {
            id: expect.any(String),
            title: expect.any(String),
            authorId: expect.any(String)
        },
        {
            id: expect.any(String),
            title: expect.any(String),
            authorId: expect.any(String)
        }         
    ]
}