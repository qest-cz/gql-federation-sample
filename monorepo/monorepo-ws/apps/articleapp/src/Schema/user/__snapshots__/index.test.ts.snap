// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user resolver get article of user 1`] = `
Object {
  "getUser": Object {
    "articles": Array [
      Object {
        "authorId": Any<String>,
        "id": Any<String>,
        "title": Any<String>,
      },
      Object {
        "authorId": Any<String>,
        "id": Any<String>,
        "title": Any<String>,
      },
    ],
    "id": Any<String>,
  },
}
`;
